File: CollaborationBackend/src/main/java/com/collaborate/service/FriendService.java
package com.collaborate.service;

import java.util.List;

import com.collaborate.model.Friend;
import com.collaborate.model.Users;

public interface FriendService {
	List<Users> listOfSuggestedUsers(String userid);
	void friendRequest(Friend friend);
	List<Friend> pendingRequests(String toId);
	void updatePendingRequests(Friend friend);
	List<String> listOfFriends(String userid);
}

File: CollaborationFrontend/Web/Services/friendservice.js
/**
 * 
 */
myApp.factory('FriendService',function($http){
	var friendservice={}
	var BASE_URL="http://localhost:8081/CollaborationMiddleware"
		friendservice.listOfSuggestedUsers=function()
		{
			return $http.get(BASE_URL+"/getSuggestedUsers")
		}
	friendservice.sendFriendRequest=function(toId)
	{
		return $http.get(BASE_URL+"/friendRequest/"+toId)
	}
	friendservice.pendingRequests=function()
	{
		return $http.get(BASE_URL+"/pendingRequests")
	}
	friendservice.updatePendingRequests=function(request)	//request is friend object
	{
		return $http.put(BASE_URL+"/updatePendingRequests",request)
	}
	friendservice.listOfFriends=function()
	{
		return $http.get(BASE_URL+"/listOfFriends")
	}
	return friendservice;
})